contracts/Items/DEFYForge.sol

```diff
@@ -0,0 +1,241 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.9;
+
+import "./IDEFYItems.sol";
+import "@openzeppelin/contracts/security/Pausable.sol";
+import "@openzeppelin/contracts/access/AccessControl.sol";
+
+contract DEFYForge is Pausable, AccessControl {
+    bytes32 public constant FORGER_ROLE = keccak256("FORGER_ROLE");
+
+    struct PrintJob {
+        IERC1155 itemContract;
+        address operativeAddress;
+        uint256[] inputMaterialsId;
+        uint256[] inputMaterialsAmount;
+        uint256 blueprintId;
+        uint256 outputPartId;
+        bool completed;
```

Comment:
We should also look at storing timestamps here for start time and duration in seconds